### PR TITLE
[tests-only][full-ci]added test scenarios for copy folder request

### PR DESCRIPTION
**Pull request by sushmita56**
### Description 
This PR adds `COPY` request implementation for copying `folders` between different spaces (`Personal`, `Shares Jail`, `Project`)

### Related Issue
https://github.com/owncloud/ocis/issues/4087

### CODE REVIEW

File: tests/acceptance/features/apiSpaces/copySpaces.feature [hunk 1]
```diff
@@ -279,3 +279,237 @@ Feature: copy file
...
+    And user "Brian" has shared a space "Project2" to user "Alice" with role "viewer"
+    And user "Brian" has shared a space "Project1" to user "Alice" with role "<role>"
+    When user "Alice" copies file "folder1" from space "Project1" to "folder1" inside space "Project2" using the WebDAV API
+    Then the HTTP status code should be "403"
+    And for user "Alice" the space "Project2" should not contain these entries:
+      | folder1/demo.txt |
+    Examples:
+      | role    |
+      | manager |
+      | editor  |
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/copySpaces.feature:349]:
```suggestion
    Examples:
      | role    |
      | manager |
      | editor  |
      |viewer|
```
Maybe we can check for viewer to viewer as well?

File: tests/acceptance/features/apiSpaces/copySpaces.feature [hunk 2]
```diff
@@ -279,3 +279,237 @@ Feature: copy file
...
+    Given the administrator has given "Alice" the role "Space Admin" using the settings api
+    And user "Alice" has created a space "Project" with the default quota using the GraphApi
+    And user "Alice" has created a folder "folder1" in space "Project"
+    And user "Alice" has created a folder "folder2" in space "Project"
+    And user "Alice" has uploaded a file inside space "Project" with content "some content" to "folder2/demo.txt"
+    And user "Alice" has shared a space "Project" to user "Brian" with role "viewer"
+    When user "Brian" copies folder "folder2" to "folder1/folder2" inside space "Project" using the WebDAV API
+    Then the HTTP status code should be "403"
+    And for user "Brian" the space "Project" should not contain these entries:
+      | folder1/folder2/demo.txt |
```
Comment (grgprarup) [inline tests/acceptance/features/apiSpaces/copySpaces.feature:311]: IMO These two scenarios can be merged by adding one column 'status_code' to example table.

File: tests/acceptance/features/apiSpaces/copySpaces.feature [hunk 3]
```diff
@@ -279,3 +279,237 @@ Feature: copy file
...
+    And user "Alice" has accepted share "/testshare" offered by user "Brian"
+    When user "Alice" copies folder "folder1" from space "Project" to "/testshare/folder1" inside space "Shares Jail" using the WebDAV API
+    Then the HTTP status code should be "403"
+    And for user "Alice" folder "testshare" of the space "Shares Jail" should not contain these files:
+      | folder1 |
+    Examples:
+      | role    |
+      | manager |
+      | editor  |
+      | viewer  |
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/copySpaces.feature:406]:
```suggestion
  Scenario Outline: User copies a folder from space project with different role to space shares jail with different role
    Given the administrator has given "Brian" the role "Space Admin" using the settings api
    And user "Brian" has created a space "Project" with the default quota using the GraphApi
    And user "Brian" has created folder "/testshare"
    And user "Brian" has created a folder "folder1" in space "Project"
    And user "Brian" has uploaded a file inside space "Project" with content "some content" to "/folder1/demo.txt"
    And user "Brian" has shared a space "Project" to user "Alice" with role "<space-role>"
    And user "Brian" has shared folder "/testshare" with user "Alice" with permissions "<share-role>"
    And user "Alice" has accepted share "/testshare" offered by user "Brian"
    When user "Alice" copies folder "folder1" from space "Project" to "/testshare/folder1" inside space "Shares Jail" using the WebDAV API
    Then the HTTP status code should be "<status-code>"
    And for user "Alice" folder "testshare" of the space "Shares Jail" <shouldOrNot> contain these files:
      | folder1/demo.txt |
    Examples:
      | space-role    | share-role    | status-code | shouldOrNot |
      | manager |   31 | 201 | should |
      | editor  | 31 | 201 | should|
      | viewer  | 31 | 201 | should|
      | manager |   17 | 403 | should not|
      | editor  | 17 | 403 | should not |
      | viewer  | 17 | 403 | should not |
```
Is it possible to do something like this?

**Comment by grgprarup** [comment]
@sushmita56 There are many scenarios with similar scenarios descriptions and steps. IMHO that can be merged.

File: tests/acceptance/features/apiSpaces/copySpaces.feature (hunk 2 above)
Comment (sushmita56) [reply tests/acceptance/features/apiSpaces/copySpaces.feature:311]: @grgprarup Though the last step of both scenarios is looking for the same resource, one of them expects the resource to exist and the other one doesn't. So if we merge these two scenarios into one, how to check for the resources inside the `Project` space as the expectation of both scenarios are different at the end?:thinking:

Comment (sushmita56) [reply tests/acceptance/features/apiSpaces/copySpaces.feature:311]: Got it!

**Review by grgprarup — COMMENTED** [review]
LGTM :+1: if CI is happy.

File: tests/acceptance/expected-failures-localAPI-on-OCIS-storage.md [hunk 4]
```diff
@@ -23,12 +23,6 @@ The expected failures in this file are from features in the owncloud/ocis repo.
...
-- [apiSpaces/copySpaces.feature:113](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiSpaces/copySpaces.feature#L113)
-- [apiSpaces/copySpaces.feature:114](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiSpaces/copySpaces.feature#L114)
-- [apiSpaces/copySpaces.feature:163](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiSpaces/copySpaces.feature#L163)
-- [apiSpaces/copySpaces.feature:260](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiSpaces/copySpaces.feature#L260)
-- [apiSpaces/copySpaces.feature:261](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiSpaces/copySpaces.feature#L261)
 - [apiSpaces/moveSpaces.feature:161](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiSpaces/moveSpaces.feature#L161)
 - [apiSpaces/moveSpaces.feature:162](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiSpaces/moveSpaces.feature#L162)
 - [apiSpaces/moveSpaces.feature:181](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiSpaces/moveSpaces.feature#L181)
```
Comment (SagarGi) [inline tests/acceptance/expected-failures-localAPI-on-OCIS-storage.md:28]: @sushmita56 can you also check if it is related to propfind in `Shares Jail`. if so please remove them from expected to failure and add step same as copy feature.

File: tests/acceptance/features/apiSpaces/copySpaces.feature [hunk 5]
```diff
@@ -279,3 +279,175 @@ Feature: copy file
...
+    And for user "Alice" the space "Project" <shouldOrNot> contain these entries:
+      | folder1/demo.txt |
+    Examples:
+      | role    | shouldOrNot | status-code |
+      | manager | should      | 201         |
+      | editor  | should      | 201         |
+      | viewer  | should not  | 403         |
+
+
+  Scenario Outline: User copies a folder from space personal to space shares jail with permission editor
```
Comment (SagarGi) [inline tests/acceptance/features/apiSpaces/copySpaces.feature:383]:
```suggestion
  Scenario Outline: User copies a folder from space personal to space shares jail with different permissions (role)
```

File: tests/acceptance/features/apiSpaces/copySpaces.feature [hunk 6]
```diff
@@ -279,3 +279,175 @@ Feature: copy file
...
+    And user "Brian" has uploaded file with content "testshare content" to "/testshare/testshare.txt"
+    And user "Brian" has shared folder "/testshare" with user "Alice" with permissions "<permissions>"
+    And user "Alice" has accepted share "/testshare" offered by user "Brian"
+    When user "Alice" copies file "/testshare/testshare.txt" from space "Shares Jail" to "testshare.txt" inside space "Personal" using the WebDAV API
+    Then the HTTP status code should be "201"
+    And for user "Alice" the space "Personal" should contain these entries:
+      | /testshare.txt |
+    And for user "Alice" the content of the file "/testshare.txt" of the space "Personal" should be "testshare content"
+    Examples:
+      | permissions |
```
Comment (SagarGi) [inline tests/acceptance/features/apiSpaces/copySpaces.feature:411]: make this consistent in every scenarios. somewhere its share-permission and somewhere its permission

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by SagarGi — COMMENTED** [review]
lgtm :+1:  let CI be green

File: tests/acceptance/features/apiSpaces/copySpaces.feature [hunk 7]
```diff
@@ -187,17 +187,17 @@ Feature: copy file
...
     When user "Alice" copies file "/testshare/testshare.txt" from space "Shares Jail" to "testshare.txt" inside space "Personal" using the WebDAV API
     Then the HTTP status code should be "201"
     And for user "Alice" the space "Personal" should contain these entries:
       | /testshare.txt |
     And for user "Alice" the content of the file "/testshare.txt" of the space "Personal" should be "testshare content"
     Examples:
-      | permissions |
-      | 31          |
-      | 17          |
+      | share-permission |
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/copySpaces.feature:198]: IMO it's better to change the new ones to <permission>, it'll give you few changes

**Review by SwikritiT — COMMENTED** [review]
Apart from the above little comment rest LGTM

**Review by SwikritiT — COMMENTED** [review]
LGTM

File: tests/acceptance/features/apiSpaces/copySpaces.feature [hunk 8]
```diff
@@ -249,8 +249,8 @@ Feature: copy file
     And user "Alice" has accepted share "/testshare2" offered by user "Brian"
     When user "Alice" copies file "/testshare1/testshare1.txt" from space "Shares Jail" to "/testshare2/testshare1.txt" inside space "Shares Jail" using the WebDAV API
     Then the HTTP status code should be "201"
-    And for user "Alice" the space "Shares Jail" should contain these entries:
-      | /testshare2/testshare1.txt |
+    And for user "Alice" folder "testshare2" of the space "Shares Jail" should contain these files:
+      | testshare1.txt |
```
Comment (individual-it) [inline tests/acceptance/features/apiSpaces/copySpaces.feature:253]: what is the difference between those two steps?

Comment (SagarGi) [reply tests/acceptance/features/apiSpaces/copySpaces.feature:253]: @individual-it inside `Shares jail` we have different shared resources which have their own id. And with specific resource we can do a `PROPFIND` request (with infinity depth) AFAIK. But `PROPFIND` inside `Shares jail` only support depth 0. https://github.com/owncloud/ocis/issues/4188. So for not relying to the issue to be fixed this step is used instead.

**Review by individual-it — APPROVED** [review]

**Pull request merged**